### DISCUSSION TITLE
Rebase instead of merging

### DISCUSSION THREAD

**Discussion by GravisZro** [DescentDevelopers/Descent3#228] [Ideas]
I've noticed that PRs are being merged instead of rebasing it's a problematic practice and you also get the "person X merged repo branch Y into repo Z" in the git log.  Anyway, I'm very much suggesting that the main branch use git rebase instead of git merge.

More info: https://stackoverflow.com/questions/804115/when-do-you-use-git-rebase-instead-of-git-merge#answers

**Comment by JeodC** [comment]
It doesn't look too problematic imho.

![image](https://github.com/DescentDevelopers/Descent3/assets/47716344/735d321f-6c2a-44a6-8736-cdaab609b481)

Rebasing would be more difficult to revert. Merging is the default action, and if it isn't broken, why fix it?

**Reply by GravisZro** [reply]
By problematic I meant that there is no certainty that you are merging in the latest version which means revisionist history can happen and existing patches can disappear. When you rebase this cannot happen as the patching branch must be up to date or it will conflict.

FYI, revisionist history has **already happened.** Note that https://github.com/DescentDevelopers/Descent3/pull/169 is no longer part of the git log and that entire patch is no longer part of branch `main`.

It makes me wonder what other patches have been lost because of merging with branches that are using old versions of the repo.

> if it isn't broken, why fix it?

I contend that it is broken.
